src/ui_utils.c

```diff
@@ -2097,7 +2097,9 @@ static void on_config_file_clicked(GtkWidget *widget, gpointer user_data)
 		if (g_file_test(global_file, G_FILE_TEST_EXISTS))
 			g_file_get_contents(global_file, &global_content, NULL, NULL);
 
-		document_new_file(utf8_filename, ft, global_content);
+		// open or create the document and mark it as changed if it didn't already exist
+		GeanyDocument *doc = document_new_file(utf8_filename, ft, global_content);
+		document_set_text_changed(doc, ! g_file_test(file_name, G_FILE_TEST_EXISTS));
```

Comment:
I didn't really ask to create it, but to open it.  And if there is data (actually, there shouldn't, as mentioned) it comes from somewhere already if I didn't write it myself, so why would I care about it?  For all I care it opened a file.  Agreed it's somewhat "magical", but the whole menu is, it's not merely opening the file.

Anyway, again I don't really mind, and can see your point technically, I just think it's not practically useful, and I fear getting a lot more users with modified configuration when the didn't mean to.